test/tests/menubar-2.js

```diff
@@ -0,0 +1,1086 @@
+      t.true(
+        checkmark,
+        'All menitemcheckbox items should have checkmark prepended'
+      );
+    }
+  });
+
+ariaTest('Test role="group" exists', exampleFile, 'group-role', async (t) => {
+  t.plan(1);
+
+  let groups = await t.context.session.findElements(By.css(ex.groupSelector));
+
+  t.truthy(
+    groups.length,
+    '"role=group" elements should be found by selector: ' + ex.group
+  );
+});
+
+ariaTest('Test role="menuitemradio" exists with accessible name',
+  exampleFile, 'menuitemradio-role', async (t) => {
+    t.plan(22);
+
+    let items = await t.context.session.findElements(By.css(ex.menuitemradioSelector));
+
+    // Test that the elements exist
+    t.truthy(
+      items.length,
+      '"role=menuitemradio" elements should be found by selector: ' + ex.menuitemradioSelector
+    );
+
+    // Test for accessible name
+
+    for (let item of items) {
+
+      // The menuitem is not visible, so we cannot use selenium's "getText" function
+      let text = await t.context.session.executeScript(function () {
+        let el = arguments[0];
+        return el.innerHTML;
+      }, item);
+
+      t.truthy(
+        text,
+        '"role=menuitemradio" elements should all have accessible text content: ' + ex.menuitemradio
+      );
+    }
+  });
+
+ariaTest('Test tabindex="-1" on role="menuitemradio"',
+  exampleFile, 'menuitemradio-tabindex', async (t) => {
+    t.plan(21);
+    await confirmAttributeValue(t, ex.menuitemradioSelector, 'tabindex', '-1');
+  });
+
+ariaTest('Text "aria-checked" appropriately set on role="menitemradio"',
+  exampleFile, 'menuitemradio-aria-checked', async (t) => {
+
+    let menus = await t.context.session.findElements(By.css(ex.menubarMenuitemSelector));
+
+    for (let grouping of ex.radioItemGroupings) {
+
+    // Reveal the elements in the dropdown
+      await menus[grouping.menuIndex].sendKeys(Key.ARROW_DOWN);
+
+      let items = await t.context.session.findElements(By.css(grouping.itemsSelector));
+
+      // Test for the initial state of checked/not checked for all radio menuitems
+
+      for (let itemIndex = 0; itemIndex < items.length; items++) {
+
+        let selected = itemIndex === grouping.defaultSelectedIndex ? true : false;
+
+        t.is(
+          await items[itemIndex].getAttribute('aria-checked'),
+          selected.toString(),
+          'Only item ' + grouping.defaultSelectedIndex + ' should have aria-select="true" in menu dropdown items: ' + grouping.itemsSelector
+        );
+
+        let checkmark = await t.context.session.executeScript(
+          checkmarkVisible, grouping.itemsSelector, itemIndex
+        );
+        t.is(
+          checkmark,
+          selected,
+          'Only item ' + grouping.defaultSelectedIndex + ' should be selected in menu dropdown items: ' + grouping.itemsSelector
+        );
+      }
+    }
+  });
+
+// KEYS
+
+ariaTest('Key SPACE and ENTER open submenu', exampleFile, 'menubar-key-space-and-enter', async (t) => {
```

Comment:
I'd prefer this as two separate tests: one for enter, one for space. It took me a while to realize that the page reload was resetting to test space.